packages/goldsky_management/src/index.ts

```diff
@@ -0,0 +1,181 @@
+            console.log(`Creating view '${toProduce.name}' in '${this.destination.name}'`)
+            this.destination.tables.push(new View(toProduce.name, this.destination));
+            return produced;
+        }).join('\n\n');
+    }
+}
+
+
+class Setup {
+    envs: Env[];
+
+    viewTemplates: ViewTemplate[];
+
+    constructor(envs?: Env[]) {
+        this.envs = envs || [];
+        this.viewTemplates = [];
+    }
+
+    addView(...v: ViewTemplate[]) {
+        this.viewTemplates.push(...v);
+    }
+
+    produce(): string {
+        this.envs.forEach(env => env.addToProduce(...this.viewTemplates));
+        return this.envs.map(env => `-- ENV ${env.name} START\n` + env.produce() + `\n-- ENV ${env.name} END\n`).join('\n\n\n');
+    }
+}
+
+// Initiate source schemas with pre-existing Goldsky tables (dispatch for now)
+let prodSource = new Schema('subgraph');
+let stageSource = new Schema('staging');
+prodSource.registerTable('dispatch');
+stageSource.registerTable('dispatch');
+prodSource.registerTable('update');
+stageSource.registerTable('update');
+prodSource.registerTable('process');
+stageSource.registerTable('process');
+
+stageSource.registerTable('recovery');
+stageSource.registerTable('process_failure');
+
+// Destination views for both environments
+let prodDest = new Schema('production_views');
+let stageDest = new Schema('staging_views');
+
+let prodDomainMapping: Domain[] = [
+    { name: 'ethereum', domain: 6648936 },
+    { name: 'avalanche', domain: 1635148152 },
+    { name: 'evmos', domain: 1702260083 },
+    { name: 'milkomedaC1', domain: 25393 },
+    { name: 'moonbeam', domain: 1650811245 },
+    { name: 'xdai', domain: 2019844457 },
+]
+let stageDomainMapping: Domain[] = [
+    { name: 'goerli', domain: 1337 },
+    { name: 'sepolia', domain: 9999 },
+]
+
+// Initiate environments with source and destination schemas.
+// both source and destination schemas are used as a source, to allow
+// re-use of freshly created views
+let prod = new Env('production', [prodSource, prodDest], prodDest, prodDomainMapping);
+let stage = new Env('staging', [stageSource, stageDest], stageDest, stageDomainMapping);
+
+// Dummy super class that leads the dance
+let s = new Setup([prod, stage]);
+
+// Register views. (name, template, required tables/views by name)
+let disp = new ViewTemplate('decoded_dispatch', fs.readFileSync('./views/decodedDispatch.sql', 'utf8'), [`dispatch`]);
+let update = new ViewTemplate('decoded_update', fs.readFileSync('./views/decodedUpdate.sql', 'utf8'), ['update']);
+let events = new ViewTemplate('events', fs.readFileSync('./views/events.sql', 'utf8'), ['decoded_dispatch', 'decoded_update', 'process']);
+let numberMessages = new ViewTemplate('number_messages', fs.readFileSync('./views/numberMessages.sql', 'utf8'), ['events']);
+let recovery = new ViewTemplate('recovery_view', fs.readFileSync('./views/recovery.sql', 'utf8'), ['recovery']);
```

Comment:
Can we name this just `recovery`?

Reply:
I wanted to separate the view and the source table. It also simultaneously intersects with the recovery within the package (`...['recovery']);` )